api/types.proto

```diff
@@ -280,6 +280,23 @@ message ContainerStatus {
 
 	int32 pid = 2 [(gogoproto.customname) = "PID"];
 	int32 exit_code = 3;
+
+	HealthStatus health = 4;
+}
+
+// HealthStatus stores information about the container's healthcheck results.
+message HealthStatus {
+	string status = 1;
+	int32 failing_streak = 2;
```

Comment:
Engine should already have it and you don't need to calculate. 

`failing_streak` actually has little value for our case. It's designed for Engine where no action is taken on the container upon failure. But engine continues counting how many times it has been failing.